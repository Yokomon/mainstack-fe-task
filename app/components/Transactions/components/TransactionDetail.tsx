"use client";

import React from "react";
import { MdOutlineArrowOutward } from "@react-icons/all-files/md/MdOutlineArrowOutward";
import { FiArrowDownLeft } from "@react-icons/all-files/fi/FiArrowDownLeft";
import clsx from "clsx";

interface TransactionDetailProps {
  title?: string;
  subTitle?: string | "successful" | "pending";
  amount: number;
  date: string;
  type: string | "deposit" | "withdrawal";
}

const TransactionDetail: React.FC<TransactionDetailProps> = ({
  title,
  type,
  amount,
  date,
  subTitle,
}) => {
  return (
    <div className="flex items-center justify-between">
      <div className="flex items-center space-x-4">
        <button
          className={clsx({
            ["flex items-center justify-center p-4 rounded-full shadow-sm duration-300"]:
              true,
            ["bg-rose-100 hover:bg-rose-200/80"]: type === "withdrawal",
            ["bg-green-100 hover:bg-green-200/80 "]: type === "deposit",
          })}
        >
          {type === "deposit" ? (
            <FiArrowDownLeft
              size={23}
              className="text-green-500"
              data-test-id="deposit-icon"
              name="deposit-icon"
              role="img"
            />
          ) : (
            <MdOutlineArrowOutward
              size={23}
              className="text-rose-400"
              data-test-id="withdrawal-icon"
              name="withdrawal-icon"
              role="img"
            />
          )}
        </button>
        <div className="space-y-1 tracking-tight">
          <h2 className="font-medium">{title}</h2>
          <p
            className={clsx({
              ["text-gray-600 text-sm"]: true,
              ["capitalize"]:
                subTitle === "pending" || subTitle === "successful",
              ["text-green-600"]: subTitle === "successful",
            })}
          >
            {subTitle}
          </p>
        </div>
      </div>
      <div className="space-y-1">
        <h1 className="text-xl font-bold">USD {amount}</h1>
        <p className="text-sm text-right text-gray-500"> {date}</p>
      </div>
    </div>
  );
};

export default TransactionDetail;
