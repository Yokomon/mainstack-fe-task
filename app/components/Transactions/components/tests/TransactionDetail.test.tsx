import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import TransactionDetail from "../TransactionDetail";

describe("TransactionDetail Component", () => {
  const mockTransactionData = {
    title: "Test Transaction",
    type: "deposit",
    amount: 1000,
    date: "2023-01-01",
    subTitle: "successful",
  };

  it("renders without crashing", () => {
    render(<TransactionDetail {...mockTransactionData} />);
    expect(screen.getByText("Test Transaction")).toBeInTheDocument();
  });

  it("displays the correct transaction type icon", () => {
    render(<TransactionDetail {...mockTransactionData} />);
    expect(screen.getByRole("img")).toBeInTheDocument();
  });

  it("displays the correct transaction title", () => {
    render(<TransactionDetail {...mockTransactionData} />);
    expect(screen.getByText("Test Transaction")).toBeInTheDocument();
  });

  it("displays the correct transaction amount", () => {
    render(<TransactionDetail {...mockTransactionData} />);
    expect(screen.getByText("USD 1000")).toBeInTheDocument();
  });

  it("displays the correct transaction date", () => {
    render(<TransactionDetail {...mockTransactionData} />);
    expect(screen.getByText("2023-01-01")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <TransactionDetail {...mockTransactionData} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
