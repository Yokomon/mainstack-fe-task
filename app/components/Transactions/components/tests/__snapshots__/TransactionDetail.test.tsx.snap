// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionDetail Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-between"
  >
    <div
      class="flex items-center space-x-4"
    >
      <button
        class="flex items-center justify-center p-4 rounded-full shadow-sm duration-300 bg-green-100 hover:bg-green-200/80 "
      >
        <svg
          class="text-green-500"
          data-test-id="deposit-icon"
          fill="none"
          height="23"
          name="deposit-icon"
          role="img"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="23"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="17"
            x2="7"
            y1="7"
            y2="17"
          />
          <polyline
            points="17 17 7 17 7 7"
          />
        </svg>
      </button>
      <div
        class="space-y-1 tracking-tight"
      >
        <h2
          class="font-medium"
        >
          Test Transaction
        </h2>
        <p
          class="text-gray-600 text-sm capitalize text-green-600"
        >
          successful
        </p>
      </div>
    </div>
    <div
      class="space-y-1"
    >
      <h1
        class="text-xl font-bold"
      >
        USD 1000
      </h1>
      <p
        class="text-sm text-right text-gray-500"
      >
         2023-01-01
      </p>
    </div>
  </div>
</DocumentFragment>
`;
